src/components/Frame/components/Toast/Toast.scss

```diff
@@ -29,7 +32,16 @@ $Backdrop-opacity: 0.88;
 }
 
 .error {
-  background: rgba(color('red', 'dark'), $Backdrop-opacity);
+  $error-color: var(--p-text-on-dark, color('white'));
+  background: var(
+    --p-critical-action,
+    rgba(color('red', 'dark'), $Backdrop-opacity)
+  );
+  color: $error-color;
+
+  .CloseButton {
+    fill: $error-color;
+  }
```

Comment:
Since the text and `x` button color is the inverse of the background (white when on black, black when on white), and since the error remains red in both dark and light mode, we need to make the text and `x` button white in either mode.